ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ethernet_interfaces.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.filter.list_compress import list_compress
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get, get_item
+
+from .utils import UtilsMixin
+
+
+class EthernetInterfacesMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def ethernet_interfaces(self) -> dict | None:
+        """
+        Return structured config for ethernet_interfaces
+        """
+        ethernet_interfaces = {}
+        for link in self._underlay_links:
+            # common values
+            ethernet_interface = {
+                "peer": link["peer"],
+                "peer_interface": link["peer_interface"],
+                "peer_type": link["peer_type"],
+                "description": self._avd_interface_descriptions.underlay_ethernet_interfaces(link["type"], link["peer"], link["peer_interface"]),
+                "speed": link.get("speed"),
+                "shutdown": self._shutdown_interfaces_towards_undeployed_peers and not link["peer_is_deployed"],
+            }
+
+            # L3 interface
+            if link["type"] == "underlay_p2p":
+                ethernet_interface.update(
+                    {
+                        "mtu": self._p2p_uplinks_mtu,
+                        "service_profile": get(self._hostvars, "p2p_uplinks_qos_profile"),
+                        "mac_security": link.get("mac_security"),
+                        "type": "routed",
+                        "ipv6_enable": link.get("ipv6_enable"),
+                        "link_tracking_groups": link.get("link_tracking_groups"),
+                    }
+                )
+
+                # PTP
+                if get(link, "ptp.enable") is True:
+                    ptp_config = {"enable": True}
+                    ptp_profile = get(self._hostvars, "switch.ptp.profile")
```

Comment:
please cache these global vars with `cached_property` since they run for every link.
See example: https://github.com/aristanetworks/ansible-avd/pull/2322/files#diff-ef96c464acb24f9b2cb71a2b9a3210b3ab4147120fee1085b218b691c2286112R266